pkg/util/tls/tls.go

```diff
@@ -122,6 +122,32 @@ func SetupTLSWithCertManager(caManager ClientCAManager, certManager certificate.
 				Certificates: []tls.Certificate{*cert},
 				ClientCAs:    clientCAPool,
 				ClientAuth:   clientAuth,
+				VerifyPeerCertificate: func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+					if len(verifiedChains) == 0 || len(verifiedChains[0]) == 0 {
+						return nil
+					}
+
+					certificate, err := x509.ParseCertificate(rawCerts[0])
```

Comment:
From VerifyPeerCertificate comment:
> // rawCerts may be empty on the server if ClientAuth is RequestClientCert or
	// VerifyClientCertIfGiven.

Possible nil pointer dereference? or it is strictly related to the `verifiedChains` len?

Reply:
```
if len(verifiedChains) == 0 || len(verifiedChains[0]) == 0 {
	return nil
}
```
means that for not verified certificates we will return and so if we don't there is rawCerts